### PR TITLE
adding thermodynamics.jl

### LINKED ISSUE #145: Add thermodynamics.jl
**context**
We need to support thermodynamics.jl

### PR DESCRIPTION
**Pull request by bischtob**
Closes #145 .

### CODE REVIEW

File: src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl [hunk 1]
```diff
@@ -36,13 +35,10 @@ function Models.default_initial_conditions(model::Nonhydrostatic2DModel)
...
-    FT = eltype(model.parameters)
-    @unpack p_0, R_d, cp_d, cv_d, g = model.parameters
-    γ = cp_d / cv_d
+    FT = eltype(model.domain)
 
-    # TODO!: Replace with Thermodynamics.jl
-    pressure(ρθ::FT) where {FT} =
-        ρθ >= FT(0) ? p_0 * (R_d * ρθ / p_0)^γ : FT(NaN)
+    # relevant parameters 
+    g::FT = CP.Planet.grav(model.parameters)
```
Comment (jiahe23) [inline src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl:41]: Can we define the const CP and TD in this file, or simply use the full package name? It took me a while to figure out what they are and where they are defined.

File: src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl [hunk 2]
```diff
@@ -103,42 +99,46 @@ function Models.make_ode_function(model::Nonhydrostatic2DModel)
         ρuh = Ym.ρuh
         ρw = Ym.ρw
         ρθ = Ym.ρθ
-
+    
+        # calculate relevant thermodynamic quantities
+        ts = @. TD.PhaseDry_ρθ(model.parameters, ρ, ρθ / ρ)
```
Comment (jiahe23) [inline src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl:104]: I think all these function in Thermodynamics.jl is dispatched upon an abstract type defined in CLIMAParameters. Do we want to do something similar in ClimaAtmos (e.g., specify the type in the Model struct )?

File: test/test_cases/initial_conditions/dry_rising_bubble_2d.jl [hunk 3]
```diff
@@ -14,12 +14,21 @@
...
-function init_dry_rising_bubble_2d(params, thermovar = :ρθ)
-    FT = eltype(params)
-    @unpack x_c, z_c, r_c, θ_b, θ_c, p_0, cp_d, cv_d, R_d, g = params
+function init_dry_rising_bubble_2d(::Type{FT}, params; thermovar = :ρθ) where {FT}
+    # physics parameters
+    p_0::FT = CLIMAParameters.Planet.MSLP(params)
+    cp_d::FT = CLIMAParameters.Planet.cp_d(params)
+    cv_d::FT = CLIMAParameters.Planet.cv_d(params)
+    R_d::FT = CLIMAParameters.Planet.R_d(params)
+    g::FT = CLIMAParameters.Planet.grav(params)
```
Comment (jiahe23) [inline test/test_cases/initial_conditions/dry_rising_bubble_2d.jl:24]: I think this looks nice. We don't need the abbrev CP in the other file ;)

File: src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl (hunk 1 above)
Comment (bischtob) [reply src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl:41]: Ok

File: src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl (hunk 2 above)
Comment (bischtob) [reply src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl:104]: I think so

File: test/test_cases/initial_conditions/dry_rising_bubble_2d.jl (hunk 3 above)
Comment (bischtob) [reply test/test_cases/initial_conditions/dry_rising_bubble_2d.jl:24]: Haha ok!

**Comment by bischtob** [comment]
bors try

**Review by jiahe23 — APPROVED** [review]

**Pull request merged**